dsgrid/query/models.py

```diff
@@ -431,6 +439,15 @@ def handle_dimension_filters(cls, dimension_filters):
                 dimension_filters[i] = make_dimension_filter(dimension_filter)
         return dimension_filters
 
+    @validator("column_type")
+    def check_column_type(cls, column_type, values):
+        if column_type == ColumnType.DIMENSION_TYPES and values["supplemental_columns"]:
+            # This would cause duplicate column names.
+            raise ValueError(
+                f"column_type={ColumnType.DIMENSION_TYPES} is incompatible with supplemental_columns"
```

Comment:
Basically because all base dimensions would be retained by default, so can't have supplemental_columns. Can't swap a base for a supplemental in this case?

Reply:
1. Specify `dimension_query_name = "state"` to get state resolution in the derived dataset: **allowed**
2. Specify `dimension_query_name = "county` as well as `supplemental_columns = ["state"]`: **not allowed** because there will be two geography columns

Reply:
Thanks, I understand it in the context of column_type == ColumnType.DIMENSION_TYPES

For my understanding, do we allow more than 1 col for a dimension in column_type == ColumnType.DIMENSION_QUERY_NAMES context? I know for derived datasets, we don't allow query_names

If so, we allow the following:
- `dimension_query_name = "state" + supplemental_columns = ["county"]`: this would be invalid to me, do we catch that?
- `dimension_query_name = "state" + supplemental_columns = ["census_division"]`

Reply:
We will silently ignore the first case and just give you "state." Is that a bug? Probably. We could check that any supplemental column is a supplemental dimension.

We add the supplemental columns before aggregating to the supplemental dimension. As long as the supplemental columns are bigger than the primary dimension, the second case should be OK. Specifying the primary as `census_division` and `supplemental_columns = ["state"]` is invalid and we don't catch it.

Reply:
Yeah, adding a no-base dim check for `supplemental_columns` seem simple enough. The rest seems harmless for now.